### ISSUE TITLE
Refactoring: README.md Needs to be streamlined

### ISSUE THREAD

**Issue by aamfahim** [SychAndrii/infusion#8]
I found the README.md hard to follow. These are the following issues I have face: 

1. I use WSL for as my terminal. I had to spend considerable amount of time in the installation section. There are a couple of perquisites to follow that should be mentioned on the .md file.
   - You have to have the latest python version installed. Need some sort of documentation link to python installation should be included.
   - I also had to use cmd aka windows environment instead of WSL since my WSL didnt have the correct configuration. But that is a me issue and I think clarifying the use of windows cmd terminal would be helpful for novice users. 
![e1](https://github.com/user-attachments/assets/4cc6fea2-cf98-4fbc-96ac-8f76cb38e037)

   - This command `python -m pipenv shell` needed a `--python` flag as well where I had to specify the path of my `python.exe` installed on windows. Which was  also difficult to find. so some sort of doc link to find python.exe path would be nice.
P.S. I think you should look into dockizing this tool possibly.
2. A large section of the .md file is dedicated to a example file. It would be better to include it in a directory in the repo and only keep essential information on the README.md and update the docs accordingly would be a better approach.
3. I have to use OpenAI API key so a link to a resource on how to get it would  be a great addition to the README.md.
4. In the Options sections short version of the `--output` and `--help` flag are missing a hyphen.

**Comment by SychAndrii** [comment]
Thank you!
I have decided not to go with dockerizing my app, because then it would not really feel like a true CLI app, and instead decided to have shell scripts for setting up all the dependencies and the correct python version. I have to say, after implementing all of your suggestions, the README file has become much easier to understand, so I appreciate the feedback! You can see all the changes I did in the pull requests.
Fixed with #15 and #14

**Comment by aamfahim** [comment]
Looks mint bruh! Instructions are LOT more clear now. Also no need to dockerize since you have added installation script for all the common environments.
